sct-commons/src/main/java/org/lfenergy/compas/sct/commons/DoTypeService.java

```diff
@@ -25,4 +32,66 @@ public Optional<TDOType> findDoType(TDataTypeTemplates tDataTypeTemplates, Predi
         return getFilteredDoTypes(tDataTypeTemplates, tdoTypePredicate).findFirst();
     }
 
-}
+    public List<DoLinkedToDa> getAllSDOLinkedToDa(TDataTypeTemplates dtt, TDOType tdoType, DoLinkedToDa filtredDoLinkedToDa) {
+        List<DoLinkedToDa> result = new ArrayList<>();
+        // DA -> BDA -> BDA..
+        sdoOrDAService.getDAs(tdoType).forEach(tda -> {
+            DoLinkedToDa doLinkedToDa = DoLinkedToDa.copyFrom(filtredDoLinkedToDa);
+            doLinkedToDa.getDataAttribute().setDaName(tda.getName());
+            if(tda.isSetFc()) {
+                doLinkedToDa.getDataAttribute().setFc(tda.getFc());
+            }
+
+            // STRUCT type (BType=STRUCT) refer to BDA, otherwise it is DA
+            if(TPredefinedBasicTypeEnum.STRUCT.equals(tda.getBType())) {
+                daTypeService.findDaType(dtt, tda.getType())
+                        .ifPresent(nextDaType -> result.addAll(getDaLinkedToBDA(dtt, nextDaType, doLinkedToDa).toList()));
+            } else {
+                doLinkedToDa.setDataAttribute(updateDataAttributeFromDaOrBda(tda, doLinkedToDa.getDataAttribute()));
+                result.add(doLinkedToDa);
+            }
+        });
+        // SDO -> SDO -> SDO..
+        sdoOrDAService.getSDOs(tdoType)
+                .forEach(tsdo -> {
+                    if(tsdo.isSetType()){
+                        findDoType(dtt, tdoType1 ->  tdoType1.getId().equals(tsdo.getType()))
+                            .ifPresent(nextDoType -> {
+                                DoLinkedToDa newDoLinkedToDa = DoLinkedToDa.copyFrom(filtredDoLinkedToDa);
+                                newDoLinkedToDa.getDataObject().getSdoNames().add(tsdo.getName());
+                                if(nextDoType.isSetCdc()) {
+                                    newDoLinkedToDa.getDataObject().setCdc(nextDoType.getCdc());
+                                }
+                                result.addAll(getAllSDOLinkedToDa(dtt, nextDoType, newDoLinkedToDa));
+                            });
+                    }
+                });
+        return result;
+    }
+    private Stream<DoLinkedToDa> getDaLinkedToBDA(TDataTypeTemplates dtt, TDAType tdaType1, DoLinkedToDa filtredDoLinkedToDa) {
```

Comment:
`filtredDoLinkedToDa` is used as a template, not a filter.
I suggest `doLinkedToDaTemplate`